### PR TITLE
Add method to get leaves of a NodeTrie

### PR DESCRIPTION
**Pull request by ajreynol**
From the oracles branch.

### CODE REVIEW

File: src/expr/node_trie.h [hunk 1]
```diff
@@ -92,6 +92,14 @@ class NodeTemplateTrie
   void clear() { d_data.clear(); }
   /** Is this trie empty? */
   bool empty() const { return d_data.empty(); }
+  /**
+   * Get leaves at the given depth, where depth>0. This argument is necessary
+   * since we do not know apriori the depth of where data occurs.
+   *
+   * If this trie stores applications of a function f, then depth should be set
+   * to the arity of f.
```
Comment (HanielB) [inline src/expr/node_trie.h:100]: What is the behavior when depth is passed wrong?

Comment (ajreynol) [reply src/expr/node_trie.h:100]: The returned vector will be inaccurate.

Comment (HanielB) [reply src/expr/node_trie.h:100]: Please add that to the documentation. (If I understand correctly doing an assertion would not really be useful)

Comment (ajreynol) [reply src/expr/node_trie.h:100]: Right, an assertion isn't possible, as the class doesn't insist on a fixed depth.

**Review by HanielB — APPROVED** [review]

**Pull request merged**